src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RSAPKCS1SignatureDeformatter.cs

```
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography
+{
+    public class RSAPKCS1SignatureDeformatter : AsymmetricSignatureDeformatter
+    {
+        private RSA _rsaKey;
+        private string _algName;
+
+        public RSAPKCS1SignatureDeformatter() { }
+        public RSAPKCS1SignatureDeformatter(AsymmetricAlgorithm key)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _rsaKey = (RSA)key;
+        }
+
+        public override void SetKey(AsymmetricAlgorithm key)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _rsaKey = (RSA)key;
+        }
+
+        public override void SetHashAlgorithm(string strName)
+        {
+            try
+            {
+                // Verify the name
+                Oid.FromFriendlyName(strName, OidGroup.HashAlgorithm);
+
+                // Keep the raw strName as Oid may change the case ("SHA1" to "sha1") making it incompatible.
```

Right... the "doesn't work later on..." is (probably) a bug due to having a switch on non-normalized inputs.  One concern I have with this is if someone passes "sha1" here we'll validate it, then let it fail later.